### ISSUE TITLE
Panic when taxonomy plural name and its content directory name are mixed cased

### ISSUE THREAD

**Issue by TheAnachronism** [gohugoio/hugo#11758]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended linux/amd64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew
</pre>

### Does this issue reproduce with the latest release?
Yes

---

I've found that for certain custom taxonomies I've defined, adding an `_index.md` file for more information, causes the build, as well as the server mode to crash:

```log
$ hugo --logLevel debug
tart building sites … 
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended linux/amd64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew

INFO  copy static: syncing static files to /
INFO  build: running step process duration 40.621283ms
INFO  build: running step assemble duration 175.913598ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x207f20a]

goroutine 53 [running]:
github.com/gohugoio/hugo/hugolib.(*pageState).getLayoutDescriptor.func1()
        github.com/gohugoio/hugo/hugolib/page.go:454 +0xea
sync.(*Once).doSlow(0x0?, 0x284270b?)
        sync/once.go:74 +0xbf
sync.(*Once).Do(...)
        sync/once.go:65
github.com/gohugoio/hugo/hugolib.(*pageState).getLayoutDescriptor(0xc000bf9240)
        github.com/gohugoio/hugo/hugolib/page.go:443 +0x73
github.com/gohugoio/hugo/hugolib.(*pageState).resolveTemplate(0xc000bf9240, {0x0, 0x0, 0xc0038697a0?})
        github.com/gohugoio/hugo/hugolib/page.go:481 +0x1c6
github.com/gohugoio/hugo/hugolib.pageRenderer(0xc0003064e0?, 0xc0009c9340, 0xc000306570?, 0xc000307290?, 0xc00073cf98?)
        github.com/gohugoio/hugo/hugolib/site_render.go:129 +0x106
created by github.com/gohugoio/hugo/hugolib.(*Site).renderPages in goroutine 16
        github.com/gohugoio/hugo/hugolib/site_render.go:77 +0x107
```

My definition of a taxonomy in `hugo.yaml`:
```yaml
taxonomies:
  category: categories
  ...

  restrictedCategories: restrictedCategories
  ...
```

To make sure it's not something weird I did in my larger hugo site, I created a new one, added the ananke theme, defined these taxonomies, and then created the _index.md file for a `restrictedCategory`.

The exact same panic happened.

What is going on here?

**Comment by TheAnachronism** [comment]
For further detail, what I did to reproduce:

1. Create new site:
```bash
hugo new site .
git init
git submodule add https://github.com/theNewDynamic/gohugo-theme-ananke.git themes/ananke
```

2. Set these taxonomies:
```yaml
taxonomies:
  tag: tags
  category: categories

  restrictedCategory: restrictedCategories
```
3. Create a post with a `restrictedCategory`:

`/content/posts/test.md`:
```markdown
---
title: 'Test'
date: 2023-11-28T22:22:33+01:00
draft: true
restrictedCategories: [
    "asdf"
]
---
```

4. Create new `_index.md` for the `restrictedCategories`:
```bash
touch content/restrictedCategories/_index.md
```

5. Hugo server and general build process crashes:
```log
$ hugo --logLevel debug
Start building sites …
hugo v0.120.4-f11bca5fec2ebb3a02727fb2a5cfb08da96fd9df+extended linux/amd64 BuildDate=2023-11-08T11:18:07Z VendorInfo=brew

INFO  copy static: syncing static files to /
INFO  build: running step process duration 199.24µs
INFO  build: running step assemble duration 282.79µs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x207f20a]

goroutine 54 [running]:
github.com/gohugoio/hugo/hugolib.(*pageState).getLayoutDescriptor.func1()
        github.com/gohugoio/hugo/hugolib/page.go:454 +0xea
sync.(*Once).doSlow(0x6?, 0xc0004d72a0?)
        sync/once.go:74 +0xbf
sync.(*Once).Do(...)
        sync/once.go:65
github.com/gohugoio/hugo/hugolib.(*pageState).getLayoutDescriptor(0xc0007e8b40)
        github.com/gohugoio/hugo/hugolib/page.go:443 +0x73
github.com/gohugoio/hugo/hugolib.(*pageState).resolveTemplate(0xc0007e8b40, {0x0, 0x0, 0x0?})
        github.com/gohugoio/hugo/hugolib/page.go:481 +0x1c6
github.com/gohugoio/hugo/hugolib.pageRenderer(0x0?, 0xc0008f3600, 0x0?, 0x0?, 0x0?)
        github.com/gohugoio/hugo/hugolib/site_render.go:129 +0x106
created by github.com/gohugoio/hugo/hugolib.(*Site).renderPages in goroutine 43
        github.com/gohugoio/hugo/hugolib/site_render.go:77 +0x107
```

**Comment by jmooring** [comment]
Workaround... lowercase

```text
touch content/restrictedcategories/_index.md
```

**Comment by TheAnachronism** [comment]
Yes but my changes aren't then accepted.
Adding for example some content to that _index.md file and using the `.Content` field in the `tems.html` template, results in no changes.
That's why I thought, that for some reason this was case-sensitive.

**Comment by TheAnachronism** [comment]
A nice!
Yeah, that works.

Any idea, why this crashes when using some unusual casing?
It seems like it shouldn't make any problems…

**Comment by jmooring** [comment]
Yeah, this is a bug. Some Hugo behaviors are case-sensitive, others are not.

**Comment by TheAnachronism** [comment]
Huh, OK.
Well, thanks for helping to get this to work. It might be worth adding to the docs that cases can play a role at certain points, if it won't be fixed in the near future.

**Comment by jmooring** [comment]
Reproducible example:

```text
git clone --single-branch -b hugo-github-issue-11758 https://github.com/jmooring/hugo-testing hugo-github-issue-11758
cd hugo-github-issue-11758
hugo
```

content structure

```text
content/
├── aBc/
│   └── _index.md
└── _index.md
```

config

```
[taxonomies]
aBc = 'aBc'
```

**Comment by khayyamsaleem** [comment]
tried looking into this, but got a little bit in the weeds and not sure how to proceed.

started with a failing test:

```
// taxonomies_test.go

func TestTaxonomiesMixedCase(t *testing.T) {
	t.Parallel()
	c := qt.New(t)

	taxonomies := make(map[string]string)
	taxonomies["aBc"] = "aBc"

	cfg, fs := newTestCfg()

	cfg.Set("taxonomies", taxonomies)
	configs, err := loadTestConfigFromProvider(cfg)
	c.Assert(err, qt.IsNil)

	const pageContent = `---
title: "Shouldn't Panic"
---
some text.`

	writeSource(t, fs, filepath.Join("content", "_index.md"), pageContent)
	writeSource(t, fs, filepath.Join("content", "aBc", "_index.md"), pageContent)

	s := buildSingleSite(t, deps.DepsCfg{Fs: fs, Configs: configs}, BuildCfg{})

	c.Assert(s, qt.IsNotNil)

}
```


This panics as above, and digging in, looks like at some point, the `viewInfo` for a contentNode associated with `content/aBc/_index.md` is `nil`. 

After an hour of trawling, I believe it has something to do with how config is loaded. Maybe the child keys of the `taxonomies` config field shouldn't be lowercased? They're lowercased automatically and indiscriminately as part of the config loading flow, deep in `hugo/common/maps/params.go:316`, called upstack (eventually) by `hugo/config/defaultConfigProvider.go:172`.

Questions:

- Is this config loading the root cause? Not sure how the `contentNode.viewInfo` gets populated, as it seems the tree-building phase of the build is over my head.
- Is the taxonomy configuration the root cause of the problem? and how `.singular` gets set as `abc` but `.plural` is set as `aBc`?

**Comment by jmooring** [comment]
This is fixed in v0.123.0.
